tools/c7n_index/c7n_index/metrics.py

```diff
@@ -143,20 +159,33 @@ def __init__(self, config, **kwargs):
 
         kwargs['port'] = config['indexer'].get('port', 9200)
 
-        self.client = Elasticsearch(
+        self.client = elasticsearch.Elasticsearch(
             host,
             **kwargs
         )
 
-    def index(self, points):
-        for p in points:
-            p['_index'] = self.config['indexer']['idx_name']
-            p['_type'] = self.es_type
-
-        results = helpers.streaming_bulk(self.client, points)
-        for status, r in results:
-            if not status:
-                log.debug("index err result %s", r)
+    def index(self, data):
+        # for backward compatibility with indexing docs from s3
```

Comment:
hmmm.. this is a fairly significant interface change, and the internal compatiblity feels a bit odd. its effectively trying to use the same interface to pass two entirely different data sets  through the same interface and then do compatibility in the implementation, ideally that would be responsibility passed to the invoker and this would be minimal here, also there's an implicit assumption on the sqs conforming to a particular data interface which isn't explicitly state or captured (ie. its notification action transport indexing) not a replay of the s3 data through sqs

Reply:
Would it be a good idea to create two different class methods, one for s3-indexing and the other for sqs? 
Regarding sqs confirming to a particular data interface, I can specify that in a Readme file.

Reply:
@kapilt  @ewbankkit ^